OWL2/java/de/unibremen/informatik/OWL2Parser.java

```diff
@@ -402,32 +405,92 @@ void renderAsOmn(OWLOntology onto) {
 		}
 
 		void renderAsXml(OWLOntology onto) {
+			FileOutputStream fos = null;
+			FileInputStream fis = null;
+			OutputStreamWriter out = null;
+			InputStreamReader in = null;
+			char[] buf = new char[8192];
+			int c, n;
+			int chomp = -1;
 			try {
 				OWLXMLRenderer xmlren = new OWLXMLRenderer();
-				File tempFile = File.createTempFile("owlTemp_1", ".xml");
-				FileWriter buf = new FileWriter(tempFile);
+				String t = System.getenv("HETS_GZIP");
+				boolean gz = t == null || t.equalsIgnoreCase("on");
+				t = gz ? ".xml.gz" : ".xml";
+				String tmpDir = System.getenv("TMPDIR");
+				File tempFile = File.createTempFile("owlTemp_1", t,
+					tmpDir == null ? null : new File(tmpDir));
+				fos = new FileOutputStream(tempFile);
+				out = new OutputStreamWriter(gz
+					? new GZIPOutputStream(fos, 4096)
+					: fos,
+					"UTF-8");
 				if (quick) {
 					onto.getOWLOntologyManager()
 						.removeAxioms(onto,onto.getAxioms());
 				}
-				xmlren.render(onto, buf);
-				_close(buf);
-				BufferedReader rBuf =
-					new BufferedReader(new FileReader(tempFile));
-				rBuf.readLine(); // ignore the first line containing <?xml version="1.0"?>
-				while (rBuf.ready()) {
-					append(rBuf.readLine()).append("\n");
+				xmlren.render(onto, out);
+				out.close();
+				out = null;
+				fos = null;
+				fis = new FileInputStream(tempFile);
+				in = new InputStreamReader(gz
+					? new GZIPInputStream(fis, 4096)
+					: fis,
+					"UTF-8");
+				// ignore the first line containing <?xml version="1.0"?>
+				while ((chomp != -2) && (c = in.read(buf, 0, 8192)) != -1) {
+					if (chomp == -1) {
+						for (int i=0; i < c; i++) {
+							if (buf[i] == '\n' || buf[i] == '\r') {
+								chomp = i;
+								break;
+							}
+						}
+					}
+					if (chomp != -1) {
+						// chomp trailing LF and CR
+						n = chomp;
+						chomp = 0;
+						for (int i=n; i < c; i++) {
+							if (buf[i] != '\n' && buf[i] != '\r') {
+								write(buf, i, c - i);
+								chomp = -2;
+								break;
```

Comment:
instead of last `for` loop: `while (buf[n] == '\n' || buf[n] == '\r') { n++ }; write(buf, n, c - n); chomp = -2;`

Reply:
Checking later, but wondering, whether stripping off <CR><LF> is really needed. If not, it can be simplified/speedup even more ...

My guess is, that the implementor of the previous solution has chosen to use a BufferedReader, because it was easiest way for him/her to do it, but was not aware of readLine()s  EOL chomping ...

Reply:
Couldn't resist ;-): Actually the while loop would need a termination condition to avoid overflows (n >= c) which is than more or less equiv. with a for loop. Some people say, for loops are faster than while (I'm not sure, think it depends), however, since it has its local var (i), VM can probably optimize the loop a little bit better (analog to C compilers, which would try to keep the value in a specific register ) ...

Furthermore, if the while loop was terminated, one needs to check, why => more redundant code (could be, that n == c but no non-EOL has been found yet) => therefore it is IMHO better to put the condition into the loop.

Anyway, the alternative would be
```
while (n < c && buf[n] == '\n' && buf[n] == '\r') { n++; }
if (n < c) { 
    write(buf, n, c - n);
    chomp = -2;
}
```
Do you prefer this one (and similar for the "chomp == -1" loop above) ?

Reply:
yes, I think this one is easier digestable (but you are right, the extra condition is needed).